Source/x11/XGServerEvent.m

```diff
@@ -511,14 +587,30 @@ - (void) processEvent: (XEvent *) event
         [self setLastTime: xEvent.xbutton.time];
         if (xEvent.xbutton.button == generic.lMouse)
           {
-            eventType = NSLeftMouseUp;
-            buttonNumber = generic.lMouse;
+            if (swapMouseButtons)
```

Comment:
When menuButtonEnabled is set to no, we could get release events without corresponding press events before them. We should add the check from above here as well.

Reply:
You're right. Thanks. Fixed.